index.src.html

```diff
@@ -2839,55 +2920,56 @@
     1. Assign `null` to {{VideoFrame/[[resource reference]]}}.
     2. Assign `true` to {{VideoFrame/[[detached]]}}.
     3. Assign `""` to {{VideoFrame/format}}.
-    4. Assign `null` to {{VideoFrame/planes}}.
-    5. Assign `0` to {{VideoFrame/codedWidth}}, {{VideoFrame/codedHeight}},
-        {{VideoFrame/cropLeft}}, {{VideoFrame/cropTop}},
-        {{VideoFrame/cropWidth}}, {{VideoFrame/cropHeight}},
-        {{VideoFrame/displayWidth}}, and {{VideoFrame/displayHeight}}.
-    6. Assign `null` to {{VideoFrame/duration}} and {{VideoFrame/timestamp}}.
+    4. Assign `0` to {{VideoFrame/[[coded width]]}},
+        {{VideoFrame/[[coded height]]}}, {{VideoFrame/[[visible left]]}},
+        {{VideoFrame/[[visible top]]}}, {{VideoFrame/[[visible width]]}},
+        {{VideoFrame/[[visible height]]}}, {{VideoFrame/[[display width]]}},
+        and {{VideoFrame/[[display height]]}}.
+    5. Assign `null` to {{VideoFrame/[[duration]]}} and
+        {{VideoFrame/[[timestamp]]}}.
 
 ### Algorithms ###{#videoframe-algorithms}
   <dfn>Create a VideoFrame</dfn> (with |output|, |timestamp|, |duration|, |displayAspectWidth|, and |displayAspectHeight|)
-  1. Let |planes| be a sequence of {{Plane}}s containing the decoded
-      video frame data from |output|.
-  2. Let |pixelFormat| be the {{PixelFormat}} of |planes|.
-  3. Let |init| be a {{VideoFramePlaneInit}} with the following
-      keys:
-      1. Assign |timestamp| to {{VideoFrameInit/timestamp}}.
-      2. Assign |duration| to {{VideoFrameInit/duration}}.
-      3. Let {{VideoFramePlaneInit/codedWidth}} and
-          {{VideoFramePlaneInit/codedHeight}}
-          be the width and height of the decoded video frame |output| in
-          pixels, prior to any cropping or aspect ratio adjustments.
-      4. Let {{VideoFramePlaneInit/cropLeft}},
-          {{VideoFramePlaneInit/cropTop}},
-          {{VideoFramePlaneInit/cropWidth}}, and
-          {{VideoFramePlaneInit/cropHeight}}
-          be the crop region of the decoded video frame |output| in
-          pixels, prior to any aspect ratio adjustments.
-      5. Let |displayWidth| and |displayHeight| be the display size of
-          the decoded frame in pixels.
-      6. If |displayAspectWidth| and |displayAspectHeight| are provided,
+  1. Let |frame| be a new {{VideoFrame}}, constructed as follows:
+      1. Assign `false` to {{VideoFrame/[[detached]]}}.
+      2. Let |resource| be the [=media resource=] described by |output|.
```

Comment:
I'd say it as "resource ownership is shared by all the frames that reference the resource. it may additionally be shared by a codec that is using/emitting the resource"

But I worry I maybe misunderstand your intent? We document the sharing in the Memory Model section. We also have this note in decode():

```
NOTE: Authors should call close() on output VideoFrames immediately when frames are no longer needed. The underlying media resources are owned by the VideoDecoder and failing to release them (or waiting for garbage collection) may cause decoding to stall.
```

LMK if more is needed, will send a quick follow up.